src/main/java/core/basesyntax/dao/animal/AnimalDaoImpl.java

```diff
@@ -12,11 +14,35 @@ public AnimalDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Animal save(Animal animal) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(animal);
+            transaction.commit();
+            return animal;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't save animal", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public List<Animal> findByNameFirstLetter(Character character) {
-        return null;
+        try (Session session = sessionFactory.openSession()) {
+            return session.createQuery("from Animal a "
+                    + "where lower(a.name) like lower(:str) ", Animal.class)
+                    .setParameter("str", character + "%")
+                    .getResultList();
+        } catch (Exception e) {
+            throw new RuntimeException("Can't get animals by first letter", e);
```

Comment:
```suggestion
            throw new RuntimeException("Can't get animals by first letter " + character, e);
```
the same with other classes